rundeckapp/grails-app/services/rundeck/services/ProjectService.groovy

```diff
@@ -653,20 +658,115 @@ class ProjectService implements InitializingBean, ExecutionFileProducer, EventPu
-        Client client = new Client(instanceUrl, apiToken)
-        ProjectImportStatus ret = client.importProjectArchive(iProject, file, true, options.executions,
-                options.configs, options.acls, options.scm)
+        File file = exportProjectToFile(project, framework, listener, archiveParams.toArchiveOptions(), authContext)
+
+        ProjectImportStatus ret = importArchiveToInstance(file, archiveParams, RundeckClient.builder().baseUrl(archiveParams.url).tokenAuth(archiveParams.apitoken).apiVersion(39).build())
+
         ImportResponse response = new ImportResponse(file: file, errors: ret.errors, ok: ret.getResultSuccess(),
                 executionErrors: ret.executionErrors, aclErrors: ret.aclErrors)
         listener?.done()
         response
     }
+
+    /**
+     * It returns a copy of the given map but the keys have strToPrepend as prefix and the values were converted to strings
+     * @param strToPrepend the prefix to be used in the keys of the returned map
+     * @param inputMap to copy and convert values to string and change keys
+     * @return a map with keys "strToPrepend.inputMapKey" and values "inputMapValue.toString"
+     */
+    static Map<String,String> prependStringToKeysInMap(String strToPrepend, Map<String, Object> inputMap){
+        Map<String,String> prependedMap = [:]
+        inputMap.each { String key, Object value ->
+            String newKey = "${strToPrepend}.${key}"
+            prependedMap[newKey] = value.toString()
+        }
+        return prependedMap
+    }
+
+    /**
+     * It communicates with the destiny server api to make it import the given archive
+     * @param archiveToExport file with project to export content
+     * @param exportArchiveParams metadata of the archive to export from this server
+     * @return an import status object where the import progress of the destiny server is updated
+     */
+    static ProjectImportStatus importArchiveToInstance(File archiveToExport, ProjectArchiveParams exportArchiveParams, Client<RundeckApi> rundeckClient) throws RuntimeException {
+        ProjectImportStatus response = new ProjectImportStatus()
+        response.successful = true
+        boolean importWebhookOpt = exportArchiveParams.exportComponents[WebhooksProjectComponent.COMPONENT_NAME]
+
+        Response<ProjectImportStatus> status = rundeckClient.getService().importProjectArchive(exportArchiveParams.getTargetproject(),
+                exportArchiveParams.preserveuuid?'preserve':'remove',
+                exportArchiveParams.exportExecutions,
+                exportArchiveParams.exportConfigs,
+                exportArchiveParams.exportAcls,
+                exportArchiveParams.exportScm,
+                importWebhookOpt,
+                importWebhookOpt && !Boolean.getBoolean(exportArchiveParams.exportOpts[WebhooksProjectComponent.COMPONENT_NAME]['inludeAuthTokens']),
+                Boolean.getBoolean(exportArchiveParams.exportComponents['node-wizard'] as String),
```

Comment:
`node-wizard` is not the same as the parameter "importNodesSources" for this method.  the importNodesSources param, should use the `exportArchiveParams.importNodesSources`

Reply:
Okay, now, since the node sources are always included in the archive so far (and also there's no option for this in the export to instance gui) I set the parameter to true whenever the the exportConfig parameter is true (because node source data is present in the project config)